### ISSUE TITLE
Advanced types doesn't work with Symbols

### ISSUE THREAD

**Issue by delprzemo** [Microsoft/TypeScript#30070]
**TypeScript Version:**  3.2.4

**Search Terms:** Symbols and advanced types; symbol extends; 

**Code**
```ts
let pushElement = Symbol('pushElement');

class TreeObject<T> {
    [pushElement](element: T extends boolean ? never : T) {
        // code here
    }
}

var tree = new TreeObject();
tree[pushElement](true); // should throw error
```

**Expected behavior:**
Boolean value parameter shouldn't be allowed here (according to type)

**Actual behavior:**
Boolean value parameter is allowed here, even if type points to not use it

**Playground Link:** https://www.typescriptlang.org/play/index.html#src=let%20pushElement%20%3D%20Symbol('pushElement')%3B%0D%0A%0D%0Aclass%20TreeObject%3CT%3E%20%7B%0D%0A%20%20%20%20%5BpushElement%5D(element%3A%20T%20extends%20boolean%20%3F%20never%20%3A%20T)%20%7B%0D%0A%20%20%20%20%20%20%20%20%2F%2F%20code%20here%0D%0A%20%20%20%20%7D%0D%0A%7D%0D%0A%0D%0Avar%20tree%20%3D%20new%20TreeObject()%3B%0D%0Atree%5BpushElement%5D(true)%3B%20%2F%2F%20should%20throw%20error

**Related Issues:** Not found

**Comment by yortus** [comment]
In your example, `T` is inferred as `{}` on the line `var tree = new TreeObject()`, since there are no other constraints or inference sites. You can see this by hovering over the `tree` variable in the playground.

This type `{}` does not extend `boolean`, so the `[pushElement]` method will accept any type of argument except `null` or `undefined`.

Also you need to use `const` (not `let`) on the first line for the symbol to be strongly typed.

**Comment by delprzemo** [comment]
@yortus oh, my bad! Too many coding this day and I missed that. Thanks for response
